### ISSUE TITLE
[Module Proposal]: `Chat-With-Your-Data`

### ISSUE THREAD

**Issue by aldunson** [Azure/Azure-Verified-Modules#1159]
### Check for previous/existing GitHub issues/module proposals

- [X] I have checked for previous/existing GitHub issues/module proposals.

### Check this module doesn't already exist in the module indexes

- [X] I have checked for that this module doesn't already exist in the module indexes.

### Bicep or Terraform?

Bicep

### Module Classification?

Resource Module

### Module Name

avm/res/chat-with-your-data/image-template

### Module Details

The Chat with your data Solution accelerator is a powerful tool that combines the capabilities of Azure AI Search and Large Language Models (LLMs) to create a conversational search experience. This solution accelerator uses an Azure OpenAI GPT model and an Azure AI Search index generated from your data, which is integrated into a web application to provide a natural language interface, including [speech-to-text](https://github.com/Azure-Samples/chat-with-your-data-solution-accelerator/blob/main/docs/speech_to_text.md) functionality, for search queries. Users can drag and drop files, point to storage, and take care of technical setup to transform documents. Everything can be deployed in your own subscription to accelerate your use of this technology.

### Do you want to be the owner of this module?

Yes

### Module Owner's GitHub Username (handle)

aldunson

**Comment by matebarabas** [comment]
@aldunson, thank you for proposing this module! As this is categorized as a pattern module, its approval will require a more thorough assessment. The AVM core team will get back to you soon. Thank you for your patience!

**Comment by rahalan** [comment]
@aldunson as @matebarabas is on leave, I will take over. Am I correct, that he send you a questionnaire? If so, can you please send the answers also to me as well, so we can move on in the process. Thanks, Rainer

**Comment by aldunson** [comment]
no, i don't have the questionnaire, could you send it to me please? aldunson@microsoft.com is my email

**Comment by aldunson** [comment]
- Shouldn’t this be a resource module? What makes it a pattern - e.g., does it deploy multiple resources? 
  - This solution is a pattern module because it deploys multiple resources that together provide a full solution for anyone who deploys it.  
- What is it for? What problem does it fix or provides a solution for? 
  - ‘Chat with your Data’ is and end-to-end baseline RAG pattern sample that uses Azure AI Search as a retriever. Users can deploy the solution, upload their documents, and immediately begin testing the demo against their own scenarios. There are more details on the GitHub page [here](https://github.com/Azure-Samples/chat-with-your-data-solution-accelerator).  
- What is/isn’t part of it? Which resource and/or pattern modules are planned to be leveraged in it? Provide a list of resources that would be part of the planned module. 

  1. Azure App Service 
  2. Azure Application Insights 
  3. Azure Bot 
  4. Azure OpenAI 
  5. Azure Document Intelligence 
  6. Azure Function App 
  7. Azure Search Service 
  8. Azure Storage Account 
  9. Azure Speech Service 

- Where is it coming from/what’s backing it - e.g., Azure Architecture Center (AAC), community request, customer example. Provide an architectural diagram and related documentation if possible - or a pointer to these if they are publicly available. 
  - The solution that this module solves is based on customer need. It was developed to help those in the field quickly demo the capabilities of Azure Open AI and other Azure tools.
 
- Solution Name
  - The agreed upon name is avm/ptn/ai/chat-with-your-data

**Comment by ChrisSidebotham** [comment]
Hi @aldunson,

Thanks for requesting/proposing to be an AVM module owner!

We just want to confirm **you agree to the below pages** that define what module ownership means:

- [Team Definitions & RACI](https://azure.github.io/Azure-Verified-Modules/specs/shared/team-definitions)
- [Shared Specification (Bicep & Terraform)](https://azure.github.io/Azure-Verified-Modules/specs/shared)
- [Module Support](https://azure.github.io/Azure-Verified-Modules/help-support/module-support)

Any questions or clarifications needed, let us know!

If you agree, please just **reply to this issue with the exact sentence below** (as this helps with our automation 👍):

"I CONFIRM I WISH TO OWN THIS AVM MODULE AND UNDERSTAND THE REQUIREMENTS AND DEFINITION OF A MODULE OWNER"

Thanks,

The AVM Core Team

#RR

**Comment by aldunson** [comment]
I CONFIRM I WISH TO OWN THIS AVM MODULE AND UNDERSTAND THE REQUIREMENTS AND DEFINITION OF A MODULE OWNER

**Comment by ChrisSidebotham** [comment]
Hi @aldunson ,

Thanks for confirming that you wish to own this AVM module and understand the related requirements and responsibilities!

Before starting development, please ensure ALL the following requirements are met.

**Please use the following values explicitly as provided in the [module index](https://azure.github.io/Azure-Verified-Modules/indexes/) page**:

- For your module:
  - `ModuleName` - for naming your module
  - `TelemetryIdPrefix` - for your module's [telemetry](https://azure.github.io/Azure-Verified-Modules/specs/shared/#id-sfr3---category-telemetry---deploymentusage-telemetry)
- For your module's repository:
  - Repo name and folder path are defined in `RepoURL`
  - Create GitHub teams for module owners and contributors and grant them permissions as outlined [here](https://azure.github.io/Azure-Verified-Modules/specs/shared/#id-snfr20---category-contributionsupport---github-teams-only).
  - Grant permissions for the AVM core team and PG teams on your GitHub repo as described [here](https://azure.github.io/Azure-Verified-Modules/specs/shared/#id-snfr9---category-contributionsupport---avm--pg-teams-github-repo-permissions).

Check if this module exists in the other IaC language. If so, collaborate with the other owner for consistency. 👍

You can now start the development of this module! ✅ Happy coding! 🎉

**Please respond to this comment and request a review from the AVM core team once your module is ready to be published! Please include a link pointing to your PR, once available. 🙏**

Any further questions or clarifications needed, let us know!

Thanks,

The AVM Core Team
